cmd/util/app.go

```diff
@@ -630,22 +630,34 @@ func constructAppsFromFileUrl(fileURL, appName string, labels, annotations, args
 	if err != nil {
 		return nil, err
 	}
-	for _, app := range apps {
-		if len(args) == 1 && args[0] != app.Name {
-			return nil, fmt.Errorf("app name '%s' does not match app spec metadata.name '%s'", args[0], app.Name)
-		}
-		if appName != "" && appName != app.Name {
-			app.Name = appName
+	filteredApps := make([]*argoappv1.Application, 0)
+
+	if len(args) == 1 {
```

Comment:
Using `args` and `AppName` is a complication of the function and it looks like we don't need to check `args`, from what I found, we always set `AppName` in high-level functions, I would like to simplify this by removing `args`, thoughts?